holonote/annotate/annotator.py

```diff
@@ -343,8 +343,18 @@ def get_display(self, *kdims: str | hv.Dimension) -> AnnotationDisplay:
             self._displays[element_key] = self._create_annotation_element(element_key)
         return self._displays[element_key]
 
-    def __mul__(self, other: hv.Element) -> hv.Overlay:
-        return other * self.get_element(*other.kdims)
+    def __mul__(self, other: hv.Element | hv.Layout | hv.Overlay | hv.NdOverlay) -> hv.Overlay:
+        if isinstance(other, (hv.Overlay, hv.NdOverlay)):
+            kdims, opts = other.kdims, other.opts.get().kwargs
+            if not kdims or kdims == ["Element"]:  # overlay and ndoverlay with no added kdims
+                # If no kdims in the overlay we use the first available
+                kdims = next(k for el in other.values() if (k := el.kdims))
+            return (other * self.get_element(*kdims)).opts(**opts)
+        elif isinstance(other, hv.Layout):
+            opts = other.opts.get().kwargs
+            return hv.Layout([el * self.get_element(*el.kdims) for el in other]).opts(**opts)
```

Comment:
```suggestion
        elif isinstance(other, hv.Layout):
            opts = other.opts.get().kwargs
            to_layout = []
            for el in other:
                kdims = el.kdims
                el_opts = el.opts.get().kwargs
                if not el.kdims or el.kdims == ["Element"]:
                    kdims = next(k for el in other.values() if (k := el.kdims))
                to_layout.append((el * self.get_element(*[kdim for kdim in kdims if kdim.name in self.spec])).opts(**el_opts))
            layout = hv.Layout(to_layout).opts(**opts)
            layout._max_cols = other._max_cols
            return layout
```

Otherwise, this would not handle instances where one of the components in the other layout is something like a dynamicmap created via rasterize. 

Also, this explicitly applies the `_max_cols` of the original `other`. Ideally, we would just `other.clone(data=[])` to keep the opts and attributes like _max_cols and then populate this with the new data, but I couldn't get that to work. It's unfortunate that _max_cols isn't just an opt kwarg for a Layout.